From: Sophie Brun <sophie@offensive-security.com>
Date: Wed, 21 Sep 2022 18:19:30 +0200
Subject: Force sqlite3 version to 1.4.4

version 1.5.0 breaks armel build
HOME=/tmp RUBYLIB=debian/metasploit-framework/usr/share/metasploit-framework/lib debian/generate-msfvenom-bash-completion.rb > debian/metasploit-framework/usr/share/bash-completion/completions/msfvenom
Overriding user environment variable 'OPENSSL_CONF' to enable legacy functions.
/<<PKGBUILDDIR>>/config/boot.rb:4: warning: already initialized constant GEMFILE_EXTENSIONS
/<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/config/boot.rb:4: warning: previous definition of GEMFILE_EXTENSIONS was here
/<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require': cannot load such file -- sqlite3/sqlite3_native (LoadError)
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/sqlite3-1.5.0-arm-linux/lib/sqlite3.rb:6:in `rescue in <top (required)>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/sqlite3-1.5.0-arm-linux/lib/sqlite3.rb:2:in `<top (required)>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/lib/msf/core/post/windows/packrat.rb:10:in `<top (required)>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:27:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:27:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:127:in `const_get'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:127:in `cget'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:239:in `block (2 levels) in eager_load'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:41:in `block in ls'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:27:in `each'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:27:in `ls'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:234:in `block in eager_load'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:219:in `synchronize'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:219:in `eager_load'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:318:in `each'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:318:in `eager_load_all'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application/finisher.rb:133:in `block in <module:Finisher>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `instance_exec'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `run'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/lib/ruby/3.0.0/tsort.rb:228:in `block in tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /usr/lib/ruby/3.0.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `call'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:60:in `run_initializers'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application.rb:391:in `initialize!'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/railtie.rb:207:in `public_send'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/railtie.rb:207:in `method_missing'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/config/environment.rb:4:in `<top (required)>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/lib/msfenv.rb:23:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/lib/msfenv.rb:23:in `<top (required)>'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from debian/generate-msfvenom-bash-completion.rb:3:in `<main>'
/<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require': /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/sqlite3-1.5.0-arm-linux/lib/sqlite3/3.0/sqlite3_native.so: cannot open shared object file: No such file or directory - /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/sqlite3-1.5.0-arm-linux/lib/sqlite3/3.0/sqlite3_native.so (LoadError)
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/sqlite3-1.5.0-arm-linux/lib/sqlite3.rb:4:in `<top (required)>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/lib/msf/core/post/windows/packrat.rb:10:in `<top (required)>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:27:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:27:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:127:in `const_get'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:127:in `cget'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:239:in `block (2 levels) in eager_load'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:41:in `block in ls'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:27:in `each'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:27:in `ls'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:234:in `block in eager_load'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:219:in `synchronize'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:219:in `eager_load'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:318:in `each'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:318:in `eager_load_all'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application/finisher.rb:133:in `block in <module:Finisher>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `instance_exec'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `run'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/lib/ruby/3.0.0/tsort.rb:228:in `block in tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /usr/lib/ruby/3.0.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `call'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:60:in `run_initializers'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application.rb:391:in `initialize!'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/railtie.rb:207:in `public_send'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/railtie.rb:207:in `method_missing'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/config/environment.rb:4:in `<top (required)>'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/lib/msfenv.rb:23:in `require'
	from /<<PKGBUILDDIR>>/debian/metasploit-framework/usr/share/metasploit-framework/lib/msfenv.rb:23:in `<top (required)>'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from debian/generate-msfvenom-bash-completion.rb:3:in `<main>'
make[1]: *** [debian/rules:41: override_dh_install] Error 1
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:8: binary-arch] Error 2
---
 metasploit-framework.gemspec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/metasploit-framework.gemspec b/metasploit-framework.gemspec
index 23992de..e732fac 100644
--- a/metasploit-framework.gemspec
+++ b/metasploit-framework.gemspec
@@ -197,7 +197,7 @@ Gem::Specification.new do |spec|
   # Needed by some modules
   spec.add_runtime_dependency 'rubyzip'
   # Needed for some post modules
-  spec.add_runtime_dependency 'sqlite3'
+  spec.add_runtime_dependency 'sqlite3','1.4.4'
   # required for Time::TZInfo in ActiveSupport
   spec.add_runtime_dependency 'tzinfo'
   # Needed so that disk size output isn't horrible
